### ISSUE TITLE
[prometheus] Helm upgrade failed: template: prometheus/templates/sts.yaml:210:156: executing "prometheus/templates/sts.yaml" at <.Values.server.persistentVolume.statefulSetNameOverride>: invalid value; expected bool

### ISSUE THREAD

**Issue by jmpalomar** [prometheus-community/helm-charts#3347]
### Describe the bug a clear and concise description of what the bug is.

Chart fails to upgrade.

### What's your helm version?

v3.11.3

### What's your kubectl version?

v1.24.4

### Which chart?

prometheus

### What's the chart version?

v22.3.2

### What happened?

When you try to upgrade from 22.0.0 to 22.3.2 with stateful set and volume storage enable it fails with following error:
```
Helm upgrade failed: template: prometheus/templates/sts.yaml:210:156: executing "prometheus/templates/sts.yaml" at <.Values.server.persistentVolume.statefulSetNameOverride>: invalid value; expected bool
```

### What you expected to happen?

Upgrade to go without issues.

### How to reproduce it?

Upgrade from 22.0.0 to 22.3.2 with statetul set and volume storage enabled.

### Enter the changed values of values.yaml?

server:

  persistentVolume:
    storageClass: "gp2"
    size: 16Gi

  statefulSet:
    enabled: true

### Enter the command that you execute and failing/misfunctioning.

helm upgrade -n prometheus prometheus prometheus-community/prometheus

**Comment by zanhsieh** [comment]
@jmpalomar 
Did you try as below?

The app.kubernetes.io/version label has been removed from the pod selector.

Therefore, you must delete the previous StatefulSet or Deployment before upgrading. Performing this operation will cause Prometheus to stop functioning until the upgrade is complete.

kubectl delete deploy,sts -l app.kubernetes.io/name=prometheus

**Comment by jmpalomar** [comment]
> @jmpalomar Did you try as below?
> 
> The app.kubernetes.io/version label has been removed from the pod selector.
> 
> Therefore, you must delete the previous StatefulSet or Deployment before upgrading. Performing this operation will cause Prometheus to stop functioning until the upgrade is complete.
> 
> kubectl delete deploy,sts -l app.kubernetes.io/name=prometheus

Yes I did, error has nothing to do with new labels, it fails when running sts.yaml template.

**Comment by vdovhanych** [comment]
@zeritti I just tried to install the version with your fix, but this doesn't fix the issue. Getting the same error.

**Comment by zeritti** [comment]
I was going to write that it was probably a forgotten repo update, cheers.

**Comment by vdovhanych** [comment]
Yeah didn't pull the repo update :) Thx for the fix!

**Comment by nicolasporta-xite** [comment]
Hi ! Not sure if it's related to this bug, but I've been trying to install this last version "22.4.1" since yesterday and I'm getting this error: 
`Error: YAML parse error on prometheus/templates/sts.yaml: error converting YAML to JSON: yaml: line 104: did not find expected key`
